### PR TITLE
Rm env ENABLE_RESOLUTION_NONE_TARGET_PORT and ENABLE_ENHANCED_RESOURCE_SCOPING

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by keithmattix** [comment]
There are still at least a couple of delta xDS issues in upstream envoy we haven't resolved yet. I'll grab the issue numbers in a bit but the main one I'm thinking of is relates to warming clusters

**Review by howardjohn — COMMENTED** [review]
Both of these are in compatibility-version-1.21.yaml, so we will need to deprecate that compatibility version first. Which we _can_ do: "Each compatibility version will only persist for a few releases, so eventually you will need to migrate to the new behavior. Currently, each compatibility version is intended to remain for at least two versions, though this is subject to change." (https://istio.io/latest/docs/setup/additional-setup/compatibility-versions/).

**Comment by hzxuzhonghu** [comment]
Sounds good. In order to deprecate, should remove the file compatibility-version-1.21.yaml

**Comment by ramaraochavali** [comment]
Should we keep DELTA_XDS little longer?

**Comment by hzxuzhonghu** [comment]
We can, this is a big change.

**Review by ramaraochavali — APPROVED** [review]

**Review by zirain — APPROVED** [review]

File: pkg/kube/client.go [hunk 1]
```diff
@@ -1325,10 +1324,7 @@ func FindIstiodMonitoringPort(pod *v1.Pod) int {
 	return 15014
 }
 
-// FilterIfEnhancedFilteringEnabled returns the namespace filter if EnhancedResourceScoping is enabled, otherwise a NOP filter.
+// FilterIfEnhancedFilteringEnabled returns the namespace filter .
 func FilterIfEnhancedFilteringEnabled(k Client) kubetypes.DynamicObjectFilter {
```
Comment (howardjohn) [inline pkg/kube/client.go:1328]: nit: can remove this function entirely now and just call `k.ObjectFilter()` directly

**Review by howardjohn — COMMENTED** [review]
LGTM

File: pkg/kube/client.go (hunk 1 above)
Comment (ramaraochavali) [reply pkg/kube/client.go:1328]: good call. https://github.com/istio/istio/pull/54093

**Pull request merged**